src/PostgREST/Query/SqlFragment.hs

```diff
@@ -223,7 +223,7 @@ asGeoJsonF = "json_build_object('type', 'FeatureCollection', 'features', coalesc
 
 customFuncF :: Maybe Routine -> QualifiedIdentifier -> QualifiedIdentifier -> SQL.Snippet
 customFuncF rout funcQi target
-  | (funcReturnsScalar <$> rout) == Just True = fromQi funcQi <> "(_postgrest_t.pgrst_scalar)"
+  | (funcReturnsScalar <$> rout) == Just True = "coalesce(" <> fromQi funcQi <> "(_postgrest_t.pgrst_scalar), '')"
```

Comment:
The `coalesce` was a bad approach because all the arguments have to be of the same type. This breaks when doing `coalesce(<json>, '')`  because `''` is not valid json for example.